test/env/env-ios-sim.js

```diff
@@ -0,0 +1,40 @@
+import moment from 'moment';
+import _ from 'lodash';
+import envBase from './env-base';
+import { logger } from 'appium-support';
+import platformDefinition from './ios-sim-platforms';
+
+const platforms = [];
+
+let deviceNameIndex = 0;
+let devicesRemaining;
+
+// Get a list of tuples containing [PLATFORM_VERSION, DEVICE_NAME]
+do {
+  devicesRemaining = false;
+
+  // Do a transpose of all of the device arrays so that we don't get the same
+  // OS grouped together
+  for (let [platformVersion, devices] of _.toPairs(platformDefinition)) {
+    if (devices.length > deviceNameIndex) {
+      platforms.push([platformVersion, devices[deviceNameIndex]]);
+      devicesRemaining = true;
+    }
+  }
+  deviceNameIndex++;
```

Comment:
I'm not quite sure about this algorithm. there are multiple devices array, but we use the same deviceNameIndex indexer

Reply:
What it's supposed to do is translate this:

```js
{
  '11.2': [
    'iPhone 5s Simulator', 'iPhone 6s Simulator', 'iPhone 7 Simulator', 'iPhone 8 Plus Simulator',
    'iPhone X Simulator', 'iPad Air 2 Simulator', 'iPad Pro (9.7 inch) Simulator'
  ],
  '11.1': [
    'iPhone 5s Simulator', 'iPhone 6s Simulator', 'iPhone 7 Simulator', 'iPhone 8 Plus Simulator',
    'iPhone X Simulator', 'iPad Air 2 Simulator', 'iPad Pro (9.7 inch) Simulator'
  ],
  '11': [
    'iPhone 5s Simulator', 'iPhone 6s Plus Simulator', 'iPhone 7 Simulator', 'iPhone X Simulator',
    'iPad Air 2 Simulator', 'iPad iPad Pro (12.9 inch) Simulator', 'iPad Pro (12.9 inch) (2nd generation) Simulator'
  ],
  '10.3': [
    'iPhone 5s Simulator', 'iPhone 6 Plus Simulator', 'iPhone 6 Plus Simulator', 'iPhone 6s Simulator',
    'iPhone SE Simulator', 'iPad Air 2 Simulator', 'iPad Pro (9.7 inch) Simulator'
  ],
  '10.2': [
    'iPhone 6s Plus Simulator', 'iPhone SE Simulator', 'iPhone 6s Simulator', 'iPhone 7 Plus Simulator',
    'iPad Pro (9.7 inch) Simulator', 'iPad Simulator', 'iPad Simulator'
  ],
  '10': [
    'iPhone 5 Simulator', 'iPhone 6 Plus Simulator', 'iPhone 6s Simulator', 'iPhone 7 Simulator',
    'iPad Air 2 Simulator', 'iPad Retina Simulator', 'iPad Simulator'
  ],
  '9.3': [
    'iPhone 4s Simulator', 'iPhone 6 Plus Simulator', 'iPhone 6s Simulator', 'iPhone 6 Plus Simulator',
    'iPad Air 2 Simulator', 'iPad Retina Simulator', 'iPad Simulator'
  ],
};
```

into this:

```js
[
  ['11.2', 'iPhone 5s Simulator'],
  ['11.1', 'iPhone 5s Simulator'],
  ['11', 'iPhone 5s Simulator'],
  ['10.2, 'iPhone 6s Plus Simulator'],
  ['10', 'iPhone 5 Simulator'],
  ['9.3', ''iPhone 4s Simulator],
  ['11.2', 'iPhone 6s Simulator'],
  ['11.1', 'iPhone 6 Simulator],
  ...
]
```

... and then using a daily sliding window, selects 5 different devices.

I wanted to transpose the arrays so that each day we test 5 different operating systems instead of doing _all_ 11.2 and then _all_ 11.1, etc....